sympy/core/function.py

```diff
@@ -30,7 +30,7 @@
 
 """
 
-from typing import Any, Dict as tDict, Optional, Set as tSet
+from typing import Any, Dict as tDict, Optional, Set as tSet, Tuple as tTuple, Union
```

Comment:
There is an unfortunate clash between SymPy types and _typing_ types, maybe we could rename them, e.g. `Dict` ==> `SymPyDict`, or `MathDict`. Same for `Set` and `Tuple`.

Reply:
I think that longer term the plan is not to use the typing types any more:
https://docs.python.org/3.9/whatsnew/3.9.html#pep-585-builtin-generic-types

Reply:
Oh, that's great news!